lang_java/parsing/tree-sitter-parser.js

```diff
@@ -0,0 +1,62 @@
+const Parser = require('tree-sitter');
+const Java = require('../parsing/tree-sitter-java'); 
+
+var myArgs = process.argv.slice(2);
+
+var fs = require("fs");
+const sourceCode = fs.readFileSync(myArgs[0]).toString();
+
+const parser = new Parser();
+parser.setLanguage(Java); 
+const tree = parser.parse(sourceCode);
+
+var lines = sourceCode.split("\n");
+
+function get_chars(startRow, startCol, endRow, endCol) {
+    var result = "";
+    if (startRow == endRow) {
+        result = lines[startRow].substring(startCol, endCol);
+    } else {
+        result = lines[startRow].substring(startCol);
+        var k;
+        for (k = startRow+1; k < endRow; k++) {
+            result += lines[k];
+        }
+        result += lines[endRow].substring(0,endCol);
+    }
+    return result.trim();
+}
+
+function get_pos(startRow, startCol, endRow, endCol) {
+    var result = "\"pos\": [\"startrow\": " + startRow + ", \"startcol\": " + startCol + ", \"endrow\": " + endRow + ", \"endcol\": " + endCol +"]";
+    return result;
+}
+
+var extra = ["class_body","block","interface_body","interface_declaration","class_declaration","method_declaration","program"];
+
+function traverse(obj) {
```

Comment:
Document what this is doing exactly! AFAIK it a recursion through AST of tree-sitter what skips through nodes of certain types and essentially constructs the JSON to be output right?

Reply:
It doesn't skip the nodes of certain types, it just doesn't search for their values because they're not as useful (ie class_body is literally the entire class body). It starts at a node and recursively adds the parameters to the JSON that it finds in the children of that node (recursively meaning it continues to do so for every child of every child node and so on).